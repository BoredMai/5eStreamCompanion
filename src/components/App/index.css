:local(.character-container) {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 100vh;
  justify-content: space-around;
  width: 100vw;
}

/* Battle Mode Styles */
:global(.battle-mode) {
  align-items: flex-end;
  flex-direction: column;
  justify-content: flex-start;
}
