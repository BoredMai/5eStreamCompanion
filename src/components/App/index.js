import cn from 'classnames';
import React, { Component } from 'react';

import { font8Bit } from '../../index.css';
import { characterContainer } from './index.css';
import CharacterBanner from '../CharacterBanner';
import Characters from '../../assets/characters';

export default class App extends Component {
  state = {
    characters: Characters,
  };

  renderBanner = (bannerProps, key) => {
    const { searchParams } = new URL(document.URL);
    const battleMode = searchParams.has('battleMode');

    return (
      <CharacterBanner battleMode={battleMode} key={key} {...bannerProps} />
    );
  };

  render() {
    const { characters } = this.state;
    const { searchParams } = new URL(document.URL);
    const battleMode = searchParams.has('battleMode');

    return (
      <div
        className={cn(
          { ['battle-mode']: battleMode },
          characterContainer,
          font8Bit
        )}
      >
        {characters.map(this.renderBanner)}
      </div>
    );
  }
}
