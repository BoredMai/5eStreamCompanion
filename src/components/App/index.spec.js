import React from 'react';
import { shallow } from 'enzyme';
import App from '.';

describe('<App />', () => {
  let wrapper;

  it('renders properly', () => {
    wrapper = shallow(<App />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('renders properly on battle mode', () => {
    window.history.pushState({}, '', 'http://localhost/?battleMode');
    wrapper = shallow(<App />);
    expect(wrapper.getElement()).toMatchSnapshot();
    expect(wrapper.find('.battle-mode')).toHaveLength(1);
  });
});
