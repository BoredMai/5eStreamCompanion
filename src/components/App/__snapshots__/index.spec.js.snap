// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
<div
  className="characterContainer font8Bit"
>
  <CharacterBanner
    ac={15}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 16,
        },
        Object {
          "attribute": "DEX",
          "value": 14,
        },
        Object {
          "attribute": "CON",
          "value": 16,
        },
        Object {
          "attribute": "INT",
          "value": 9,
        },
        Object {
          "attribute": "WIS",
          "value": 12,
        },
        Object {
          "attribute": "CHA",
          "value": 12,
        },
      ]
    }
    battleMode={false}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Fighter",
        },
      ]
    }
    hp={
      Object {
        "current": 13,
        "max": 13,
      }
    }
    level={1}
    name="Victor 'Invictus' Ursus"
    player="PhoenixFire"
    race="Half-Orc"
    shortName="Invictus"
    xp={200}
  />
  <CharacterBanner
    ac={10}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 17,
        },
        Object {
          "attribute": "DEX",
          "value": 8,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 13,
        },
        Object {
          "attribute": "WIS",
          "value": 10,
        },
        Object {
          "attribute": "CHA",
          "value": 16,
        },
      ]
    }
    battleMode={false}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Bard",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Savarax 'Lee' Swovarah"
    player="Chaotic"
    race="Dragonborn"
    shortName="Lee"
    xp={0}
  />
  <CharacterBanner
    ac={16}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 14,
        },
        Object {
          "attribute": "DEX",
          "value": 17,
        },
        Object {
          "attribute": "CON",
          "value": 16,
        },
        Object {
          "attribute": "INT",
          "value": 8,
        },
        Object {
          "attribute": "WIS",
          "value": 10,
        },
        Object {
          "attribute": "CHA",
          "value": 12,
        },
      ]
    }
    battleMode={false}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Barbarian",
        },
      ]
    }
    hp={
      Object {
        "current": 14,
        "max": 14,
      }
    }
    level={1}
    name="Carrie"
    player="Hyuna"
    race="Halfling"
    xp={0}
  />
  <CharacterBanner
    ac={13}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 9,
        },
        Object {
          "attribute": "DEX",
          "value": 14,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 10,
        },
        Object {
          "attribute": "WIS",
          "value": 14,
        },
        Object {
          "attribute": "CHA",
          "value": 16,
        },
      ]
    }
    battleMode={false}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Bard",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Einwynn Tordannon"
    player="Jack"
    race="Lamia"
    xp={200}
  />
  <CharacterBanner
    ac={15}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 14,
        },
        Object {
          "attribute": "DEX",
          "value": 10,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 17,
        },
        Object {
          "attribute": "WIS",
          "value": 13,
        },
        Object {
          "attribute": "CHA",
          "value": 8,
        },
      ]
    }
    battleMode={false}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Mystic",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Medic"
    player="Nate"
    race="Warforged"
    xp={200}
  />
</div>
`;

exports[`<App /> renders properly on battle mode 1`] = `
<div
  className="battle-mode characterContainer font8Bit"
>
  <CharacterBanner
    ac={15}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 16,
        },
        Object {
          "attribute": "DEX",
          "value": 14,
        },
        Object {
          "attribute": "CON",
          "value": 16,
        },
        Object {
          "attribute": "INT",
          "value": 9,
        },
        Object {
          "attribute": "WIS",
          "value": 12,
        },
        Object {
          "attribute": "CHA",
          "value": 12,
        },
      ]
    }
    battleMode={true}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Fighter",
        },
      ]
    }
    hp={
      Object {
        "current": 13,
        "max": 13,
      }
    }
    level={1}
    name="Victor 'Invictus' Ursus"
    player="PhoenixFire"
    race="Half-Orc"
    shortName="Invictus"
    xp={200}
  />
  <CharacterBanner
    ac={10}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 17,
        },
        Object {
          "attribute": "DEX",
          "value": 8,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 13,
        },
        Object {
          "attribute": "WIS",
          "value": 10,
        },
        Object {
          "attribute": "CHA",
          "value": 16,
        },
      ]
    }
    battleMode={true}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Bard",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Savarax 'Lee' Swovarah"
    player="Chaotic"
    race="Dragonborn"
    shortName="Lee"
    xp={0}
  />
  <CharacterBanner
    ac={16}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 14,
        },
        Object {
          "attribute": "DEX",
          "value": 17,
        },
        Object {
          "attribute": "CON",
          "value": 16,
        },
        Object {
          "attribute": "INT",
          "value": 8,
        },
        Object {
          "attribute": "WIS",
          "value": 10,
        },
        Object {
          "attribute": "CHA",
          "value": 12,
        },
      ]
    }
    battleMode={true}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Barbarian",
        },
      ]
    }
    hp={
      Object {
        "current": 14,
        "max": 14,
      }
    }
    level={1}
    name="Carrie"
    player="Hyuna"
    race="Halfling"
    xp={0}
  />
  <CharacterBanner
    ac={13}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 9,
        },
        Object {
          "attribute": "DEX",
          "value": 14,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 10,
        },
        Object {
          "attribute": "WIS",
          "value": 14,
        },
        Object {
          "attribute": "CHA",
          "value": 16,
        },
      ]
    }
    battleMode={true}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Bard",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Einwynn Tordannon"
    player="Jack"
    race="Lamia"
    xp={200}
  />
  <CharacterBanner
    ac={15}
    attributes={
      Array [
        Object {
          "attribute": "STR",
          "value": 14,
        },
        Object {
          "attribute": "DEX",
          "value": 10,
        },
        Object {
          "attribute": "CON",
          "value": 14,
        },
        Object {
          "attribute": "INT",
          "value": 17,
        },
        Object {
          "attribute": "WIS",
          "value": 13,
        },
        Object {
          "attribute": "CHA",
          "value": 8,
        },
      ]
    }
    battleMode={true}
    classes={
      Array [
        Object {
          "level": 1,
          "name": "Mystic",
        },
      ]
    }
    hp={
      Object {
        "current": 10,
        "max": 10,
      }
    }
    level={1}
    name="Medic"
    player="Nate"
    race="Warforged"
    xp={200}
  />
</div>
`;
